src/Bicep.Core/Semantics/SemanticModel.cs

```diff
@@ -30,13 +32,15 @@ public class SemanticModel : ISemanticModel
         private readonly Lazy<ImmutableArray<DeclaredResourceMetadata>> declaredResourcesLazy;
         private readonly Lazy<ImmutableArray<IDiagnostic>> allDiagnostics;
 
-        public SemanticModel(Compilation compilation, BicepFile sourceFile, IFileResolver fileResolver, IBicepAnalyzer linterAnalyzer)
+        public SemanticModel(Compilation compilation, BicepFile sourceFile, IFileResolver fileResolver, RootConfiguration configuration, ApiVersionProvider apiVersionProvider, IBicepAnalyzer linterAnalyzer)
```

Comment:
Compilation has RootConfiguration. Can we use it from there instead of passing it again here?

Reply:
Thanks!

Reply:
1e2d082ba